[v1.3][F11-B1] CRM-Book

davidten opened the pull request:
Changelog: This PR Contains
1. Updates to Developer Guide
2. Separating of leads and contacts
3. Adding placeholder command for LinkedIn login

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -798,17 +799,32 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |salesperson |see my calendar |schedule meetings without conflicting
+
+|`*` |salesperson |schedule a meeting using zoom |schedule meetings easily and with a link other salespeople can use
```
What is zoom? Might want to add it in the glossary

karrui commented on docs/UserGuide.adoc:
```diff
@@ -17,7 +17,7 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
 == Introduction
 
-AddressBook Level 4 (AB4) is for those who *prefer to use a desktop app for managing contacts*. More importantly, AB4 is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB4 can get your contact management tasks done faster than traditional GUI apps. Interested? Jump to the <<Qui ...
+CRM Book is for those who *prefer to use a desktop app for managing their Leads, Contacts, and Accounts*. More importantly, CRM Book is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, CRM Book can get your contact management tasks done faster than traditional GUI apps. Intereste ...
```
Can the introduction be more specific? Try to include your target user in the description.

karrui commented on src/main/java/seedu/address/logic/commands/LinkedInLoginCommand.java:
```diff
@@ -0,0 +1,32 @@
+     * Creates a LinkedInLogin log a Salesperson in to LinkedIn
+     */
+    public LinkedInLoginCommand() {
+
```
Do not merge incomplete features into the master branch of your team repo. Remember, the master branch is the gold standard -- all advertised features (that are not marked coming soon) should be working when tested. 
There should be at least a basic functionality for the command.

karrui commented on src/main/java/seedu/address/logic/commands/LinkedInLoginCommand.java:
```diff
@@ -0,0 +1,32 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+
+/**
+ * Allows a user to login to their LinkedIn account
```
Does not tell me much about this class. Be more specific.

karrui commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -35,6 +35,15 @@ public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tag
         this.tags = new UniqueTagList(tags);
     }
 
+    protected Person() {
```
Since `Person` is not used anymore, consider making it an interface so that it is more obvious this is a superclass

karrui commented on docs/UserGuide.adoc:
```diff
@@ -60,10 +60,25 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 Format: `help`
 
-=== Adding a person: `add`
+=== Log in to LinkedIn : 'linkedin_login'
```
Might want to add a `(Feature coming soon)` disclaimer since it is not working yet. Same for the others.

karrui replied:
Would the command be too troublesome to type? Remember, the aim of this application is to provide an easy to use command line like interface for the speed of performing tasks.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-F11-B1/main
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Update this line! Preferably `XXX forked from Team SE-EDU`

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -798,17 +799,32 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
```
You can remove this if there is no more to be added. Of course, keep it if you still have any. (But remember to remove it soon, you should finalise the user stories soon.)

karrui reviewed (commented):
**DO NOT CLOSE THIS PR!** 
Make changes as requested/suggested throughout the weeks until v1.5 :)
Few notes:
No new NFRs found, please add them
No new use cases found, please add them

Good to see target user profiled.
Be reminded that you will implement 1 big feature each, and it should solve the target user’s problem.
You can also define a person and give a name (to easily refer to him/her later in the doc)

Good to see that the milestones for v1.0 to v1.5 have been created.

karrui commented:
Reviewing now; but a few things to take note for next submission (i.e v1.2 onwards)

Comment here when you are ready for review and give a summary of the changes -- i.e
"Add xxx function"
"Change xxx", etc
So I can see what new changes there are :)

karrui commented on docs/AboutUs.adoc:
```diff
@@ -10,46 +10,29 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
 
```
Update the above parts too

karrui commented on docs/AboutUs.adoc:
```diff
@@ -10,46 +10,29 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
 
-Role: Project Advisor
+Role: Team Lead + Scheduling and Tracking + CRM Expert
+Responsibilities: LinkedIn Integration
```
Remember to use asciidoctor compatible code to do line breaks -- current version is in a single line

karrui commented on docs/AboutUs.adoc:
```diff
@@ -10,46 +10,29 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
-Role: Team Lead +
-Responsibilities: UI
+Role: Documentation + Deliverables and deadlines
+Responsibilities: Smart Duplicate detection + Data Enrichment + UI
```
Remember to use asciidoctor compatible code to do line breaks -- current version is in a single line

karrui commented on docs/AboutUs.adoc:
```diff
@@ -10,46 +10,29 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
-Role: Developer +
-Responsibilities: UI
+Role: Developer + Git Expert + Integration
+Responsibilities: Lead, Contact and Account Objects. Calendar and Meeting Scheduling.
```
Remember to use asciidoctor compatible code to do line breaks -- current version is in a single line

karrui commented on docs/UserGuide.adoc:
```diff
@@ -60,10 +79,24 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 Format: `help`
 
-=== Adding a person: `add`
+=== Log in to LinkedIn : 'linkedin_login' (Half implemented)
```
Do not add non-functioning functions for the user to use, as every version increase should be a fully working product. Keep in-progress code in the personal repo and try not to merge it into the team repo

karrui commented on docs/UserGuide.adoc:
```diff
@@ -60,10 +79,24 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+[TIP]
+A LinkedIn Login is required if you want to synchronize your CRM Book with your LinkedIn Account. If this is your first login, a synchronization will also take place.
+
+=== Update from LinkedIn : 'updateLinkedIn' (IN PROGRESS)
```
Try to keep command syntax the same so as to not confuse the user

karrui commented on src/main/java/seedu/address/commons/core/Oauth2Client.java:
```diff
@@ -0,0 +1,68 @@
+    public static void startServer() throws IOException {
+        int port = 13370;
+        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+        server.createContext("/test", new MyHandler());
```
Try to ensure enhancements have at least basic working functionality as discussed in the tutorial

karrui commented on src/main/java/seedu/address/logic/LogicManager.java:
```diff
@@ -35,7 +36,12 @@ public LogicManager(Model model) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         try {
-            Command command = addressBookParser.parseCommand(commandText);
+            Command command = null;
```
Did this need to be refactored? Seems to add more room for error.

karrui commented on src/main/java/seedu/address/logic/commands/ConvertCommand.java:
```diff
@@ -0,0 +1,118 @@
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
```
Remember to update header names. It returns a `Contact` from what I can tell, and not a `Person`

karrui commented on src/main/java/seedu/address/model/person/Contact.java:
```diff
@@ -0,0 +1,108 @@
+ * Represents a Person in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Contact extends Person {
```
Since `Person` is still a class and not an interface, probably do not need to repeat all methods or variables due to inheritance. 
In addition, is this class really needed? Seems to me that a simple new variable of `Type` class would suffice in the original `Person` class.

karrui commented on src/main/java/seedu/address/model/person/Lead.java:
```diff
@@ -0,0 +1,108 @@
+ * Represents a Person in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Lead extends Person {
```
Same questions as above `Contact` class

karrui commented on src/main/java/seedu/address/model/person/exceptions/PersonWrongType.java:
```diff
@@ -0,0 +1,6 @@
+/**
+ * Signals that the operation is unable to act on given type of person
+ */
+public class PersonWrongType extends Exception {}
```
Wrong format for exceptions. Exceptions should be clear that they are exceptions, and thus the recommended naming syntax could be `PersonWrongTypeException`

karrui commented on src/main/java/seedu/address/ui/BrowserWindow.java:
```diff
@@ -0,0 +1,72 @@
+     * Shows the help window.
+     * @throws IllegalStateException
+     * <ul>
+     *     <li>
```
HTML code in a Java file? No idea what this is.

karrui reviewed (commented):
@davidten @Sheikh-Umar @WoodyLau 
v1.1 reviewed, remember to not close your PR.

@Sheikh-Umar your enhancement is too minor -- why did only 3 commands receive aliases?

To all: Please update developer guide when new features are added. I don't see any documentation on how the code is implemented for the new features

**Some admin matters.**
I would want you to mention the following in your Developer guide

For each team member what are the features (major and minor) you are proposing?
Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

shumarb commented:
@karrui I wanted to do the alias commands on a weekly basis so that I am able to do a code enhancement on a weekly basis.

karrui commented:
@davidten @Sheikh-Umar @WoodyLau @zhuleyan @A0155428B Please note the new post about defining your major and minor enhancements - https://github.com/nus-cs2103-AY1718S2/forum/issues/28
Namely, a major enhancement should span multiple stories, with stories spanning multiple tasks, and a minor enhancement should be a single story. Make sure your issue tracker and milestones are properly set up before the tutorial.

A demo issue tracker has also been released for your perusal. You may also find it at https://github.com/MyTAOrg/DemoGradle/issues

@Sheikh-Umar Your alias command should not span multiple weeks. It is an extremely minor enhancement and should be all done at once. Last week's enhancement by you was not satisfactory. Please finish up the alias command by this week if you are using it as your minor enhancement.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
-=== [Proposed] Data Encryption
 
-_{Explain here how the data encryption feature will be implemented}_
+// tag::remark[]
```
Wish to see more sequence diagrams of how the command is invoked. Goes for all the other documentation. Please make sure this week's enhancement has some pictures. Remember, developer documentation has its uses (developer-as-maintainer and developer-as-user) and have to make it as readable as possible.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+}
+----
+
+The model `Person` is modified to have a new field `Remark`.
```
Hope to see the Model diagram of this updated. The current model diagram has not been updated to reflect the changes.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+** Cons: Cannot add a person with remark.
+* **Alternative 2:** Just make it a part of `EditCommand` and `AddCommand`
+** Pros: Only need to add remark field to every related class. We can add a person with remark.
+** Cons: We may have to type too many things when adding a person. Also, some people don’t have any remark.
```
This argument is invalid for `EditCommand` - the current implementation of `EditCommand` already allows for optional parameters -- you do not have to type out all the parameters if editing.

Argument is valid for `AddCommand`.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+To allow easy identification of different tags, a color tag feature is introduced.
+
+A string array is declared in PersonCard.java to include all the different colors for tags. These colors are defined, with their backgroud color and text color spelled out, in the the DarkTheme.css file.
```
Add the relevant code for more clarification of how method really works.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+Many existing commands like AddCommand have been modified to fit the new paradigm so that it only adds Leads.
+
+A new conversion command ConvertCommand has been added to convert Leads into Contacts
```
Add the relevant code and diagrams of how this is achieved. Remember to also update the Model diagram to account for this new Lead and Contact classes in the Person class.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+The linkedInLogin mechanism is handled largely by the oAuth2Client interacting with the browser window, which the user will interact with in order to give LinkedIn their username, password, as well as granting the application permission for use.
+
+When the `linkedIn_login` command is called, a `ShowBrowserRequestEvent` will be fired by `LinkedInLoginCommand(Class)` and picked up by the MainWindow. The MainWindow in turn will call the `OAuth2Client(Class)` which will fire up the browser awaiting an authorization code from LinkedIn that will be sent after the user has successfully logged in and granted the CRM Book permission.
```
Show code too.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +1097,71 @@ Use case ends.
+
+1. User enters a Lead to add
+2. CRM Book determines if Lead can be added
+3. CRM Book displays message on whether Lead entered can be added or is a duplicate
```
Wrong sequence -- MSS should be Actor-System-Actor-System

davidten replied:
@Sheikh-Umar please resolve this

shumarb replied:
I have resolved it.

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +1097,71 @@ Use case ends.
++
+[none]
+** 1a1. CRM displays error message.
++
```
It should return to the user, which step should it return to?

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +1097,71 @@ Use case ends.
++
+Use case ends.
+
+* 2a. Lead entered has a phone number or email address stored earlier.
```
Always show who is doing the action -- User or System?

karrui commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +1097,71 @@ Use case ends.
+* 2a. Desired list is empty.
+** 2a1. CRM displays message that list is empty.
++
+Use case ends.
```
Where are the other use cases for your enhancements? This is also part of documentation. @davidten @WoodyLau @zhuleyan @A0155428B

karrui commented on src/test/java/systemtests/HelpCommandSystemTest.java:
```diff
@@ -59,6 +59,10 @@ public void openHelpWindow() {
         getMainWindowHandle().focus();
 
+        // open help window and give it focus using help alias
+        executeCommand(HelpCommand.COMMAND_ALIAS);
```
This test does nothing. This is not actually testing whether the command alias works. Please try to understand what the tests are doing instead of blindly copy and pasting :(

karrui reviewed (commented):
Your current documentation still has a lot of room for improvement (both user guide and developer guide).
The architecture diagrams are outdated. Please update them. Many enhancements also have missing use cases (you are actually supposed to write out the use cases before you implement your enhancements so you have the requirements in mind).
Please put more effort into it because your documentation will be used in the grading.
 
Some other reminders:

- Please make sure that each person has written a section on implementation in the Developer Guide for their major feature.
- Tag each feature in the User Guide with (“Since v1.x”) asap.
- Remember to build a .jar file for v1.3. We will be downloading the fresh copy of it from your releases to demo in class. It must be able to run all your enhancements.
- Use UML diagrams in the Developer Guide when explaining the implementation. (There is an IntelliJ plugin called SequenceDiagram that may help.)
- Some comments that call for changes in code are still undone.

**IMPORTANT**
Please make sure to tag your repo with the correct version (e.g. for last tutorial the commits up till v1.2 should be tagged v1.2). If you are not sure how, see the textbook (W3.9c). I will be checking this as well.

This is so I can see what commits correspond to which version.

zhuleyan committed:
Userguide (#122)

* Correct userguide for remark command

* add minor feature to dev guide

* update dev guide and user guide to v1.3

* follow MSS

zhuleyan committed:
add one line to import image correctly (#127)

WoodyLau committed:
Adding editdetails to User Guide (#129)

WoodyLau committed:
Adding fields to Leads and  Contacts (#128)

shumarb committed:
Duplicate detection (#121)

* Duplicate detection based on phone and email address

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

shumarb committed:
Updated Architecture Diagram via Logic and Model Components (#130)

davidten committed:
SharePostToLinkedIn (#131)

* Save LinkedIn access_token to config.json file

* Finishes shareToLinkedIn Command

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Code credit (#132)

* Adds author tag

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
DocChange (#133)

* userguide change

* Updates aboutUs slightly

* Update Dev Guide

* Update Architecture Diagram

* updates to devDoc

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

WoodyLau committed:
Added editdetails to command summary. (#134)


* Add editdetails to command summary

zhuleyan committed:
update UserGuide for import command (#139)

* update UserGuide for import command

* change number

zhuleyan committed:
Add an ImportCommand (#140)

* import command

* checkstyle

* detect csv file

* add white space

shumarb committed:
Author Annotation and Updated MSS (#136)

* DeveloperGuide: Updated MSS for listing Leads/Contacts

* UniquePersonList: Added author annotation

* Added command word and alias command for listing leads

* UserGuide: Edited command word for listing leads

* UserGuide: Minor edit of MSS

* UserGuide: Updated command word and alias for listing Leads

* Edited MSS

A0155428B committed:
Adds Change theme command (#137)

* Adds a LightTheme and changeTheme related commands(TBC)

* implement change theme statically

* collate code and add exception

* update lightTheme list view background color

* delete theme class

* adds Theme class

* corrects style

* Corrects import order

* Corrects import order

shumarb committed:
Updated summary of display command (#142)

* DeveloperGuide: Updated MSS for listing Leads/Contacts

* UniquePersonList: Added author annotation

* Added command word and alias command for listing leads

* UserGuide: Edited command word for listing leads

* UserGuide: Minor edit of MSS

* UserGuide: Updated command word and alias for listing Leads

* Edited MSS

* Updated summary of display command

shumarb committed:
Display list of Leads and Contacts (#141)

* Edited find search to include finding a Lead

* Create display command to display all Leads

* Added new line

* Updated ordering of commands

* Updated display to display all Leads or Contacts

* Created DisplayCommand parser

* Re-positioned display command in Parser

* Edited spacing

* Update where find command searches names only

* Added author annotation

* Display command testing

* Re-positioned imports

shumarb committed:
v1.4 Documentation (#143)

* DeveloperGuide: Updated MSS for listing Leads/Contacts

* UniquePersonList: Added author annotation

* Added command word and alias command for listing leads

* UserGuide: Edited command word for listing leads

* UserGuide: Minor edit of MSS

* UserGuide: Updated command word and alias for listing Leads

* Edited MSS

* Updated summary of display command

* Updated MSS for display command

* Edited MSS for display command

* Added diagrams for DisplayCommand

* Added other's work back

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

shumarb committed:
Create collated

shumarb committed:
Delete collated

A0155428B committed:
update userguide and dev guide for change theme command
correct annotation

shumarb committed:
Create functional

shumarb committed:
Delete functional

shumarb committed:
Functional folder under Collated folder

shumarb committed:
Documents for collated/functional

shumarb committed:
Delete documents

shumarb committed:
Test folders under Collated folder

shumarb committed:
Delete test

shumarb committed:
Create document

shumarb committed:
Collated test documents

shumarb committed:
Create document

shumarb committed:
Collated unused documents

A0155428B committed:
Corrects styple for userguide and GuiTestAssert

A0155428B committed:
Documentation1.4 (#144)

* updates user story

* Updates contribution

* Adds description for color tag

* update userguide and dev guide for change theme command
correct annotation

* Corrects styple for userguide and GuiTestAssert

A0155428B committed:
Updates autocomplete for display, edit details, import, remark, sharetolinkedin

A0155428B committed:
updates autocomplete for changetheme

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

* 'master' of https://github.com/CS2103JAN2018-F11-B1/main:
  Documentation1.4 (#144)
  Collated unused documents
  Create document
  Collated test documents
  Create document
  Delete test
  Test folders under Collated folder
  Delete documents
  Documents for collated/functional
  Functional folder under Collated folder
  Delete functional
  Create functional
  Delete collated
  Create collated

A0155428B committed:
deletes liu.md

shumarb committed:
Delete document

shumarb committed:
Delete document

WoodyLau committed:
Lead contact (#148)

* Add Ui image

* Implement Leads and Contacts

* Fix style problems

* Fix style problems

* Revert syntax changes, resolve conflict probably

* Add comments, fix converting contact bugs

* Update documentation and fix bugs

* Fix style

* Fix style

* Added editdetails command, which lets user edit nonessential fields for Leads and Contacts

* Fix some style

* Fix some styles

* Fix some styles

* Fix some styles

* Fix some styles

* Added details fields to UI

* Fix changes

* Fix changes

A0155428B committed:
updates Changetheme command (#145)

* updates user story

* Updates contribution

* Adds description for color tag

* updates color tag for light theme

shumarb committed:
Updated my responsibilities (#146)

* Updated my responsibility

* Updated my responsibilities

A0155428B committed:
Removes Liu.md (#147)

* updates user story

* Updates contribution

* Adds description for color tag

* deletes liu.md

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

shumarb committed:
Add files via upload

shumarb committed:
Add files via upload

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

WoodyLau committed:
Docs (#151)

* Update responsibilities

davidten committed:
claim credit (#152)

davidten committed:
Login and share defense (#150)

* defensive methods for starting server and sharing to linkedin

WoodyLau committed:
Adding Accounts to Contacts (#149)

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

WoodyLau committed:
Comment out Alias test (#155)

* Comment out Alias test

* Fix style

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

A0155428B committed:
Search for user when clicking on user panel (#156)

A0155428B committed:
Documentation1.4 (#157)

* updates user story

* Updates contribution

* Adds description for color tag

* deletes liu.md

* delete md files created due to wrong annotation

A0155428B committed:
adds a welcome page

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

shumarb committed:
Merge pull request #159 from A0155428B/autocomplete1.4

updates autocomplete to include all commands in v1.4

shumarb committed:
Merge pull request #158 from A0155428B/changetheme1.4b

updates ui to include a welcome page

shumarb committed:
Latest collated code for functional code

shumarb committed:
Latest collated test code

shumarb committed:
Delete WoodyLau.md

shumarb committed:
Delete A0155428B.md

shumarb committed:
Delete Sheikh-Umar.md

shumarb committed:
Delete davidten.md

shumarb committed:
Delete zhuleyan.md

shumarb committed:
Create document

shumarb committed:
Add files via upload

shumarb committed:
Delete document

shumarb committed:
Delete A0155428B.md

shumarb committed:
Delete Liu.md

shumarb committed:
Delete Sheikh-Umar.md

shumarb committed:
Delete WoodyLau.md

shumarb committed:
Delete davidten.md

shumarb committed:
Delete zhuleyan.md

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Updates UserGuide (#160)

WoodyLau committed:
Move account command to 1.5 (#161)

* Update features for Woody

* Add short description of Leads and Contacts feature

* add build.gradle

* Add convert Use Case. Add collating

* Add expected commands, add feature descriptions

* Fix duplicate import

* Fix mistaken deletion

* Fix MSS

* Add editdetails to User Guide

* Add editdetails to command summary

* Update responsibilities

* Account command will be implemented in 1.5, for now use editdetails

shumarb committed:
Updated UI (#162)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

zhuleyan committed:
add acknowledgement (#177)

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

A0155428B committed:
remove comment sign for clear alias test in ClearCommandSystemTest

A0155428B committed:
Adds ChangeThemeCommandTest and ChangeThemePaserTest (#199)

* adds changeThemeCommandPaserTest and ChangeThemeCommandTest

* deletes unused import

* Corrects style

* Corrects indentation

WoodyLau committed:
Merge pull request #198 from A0155428B/ClearCommandSystemTest

Clear command system test

WoodyLau committed:
Change User Guide for clarity (#201)

* Change user guide after feedback

zhuleyan committed:
must type r/ to remove a remark (#202)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

shumarb committed:
As a salesperson, I can find a lead or contact by phone number and email address (#200)

* Edited find to make it search for email address and phone number

* Test cases for phone number and email address combinations

* Commented out test cases

* Updated testing based on Travis

* Updated testing

* Updated testing from false to true

* Removed author annotation

* Added author annotation xml version

* Added and edited test cases

* Removed trailing white space

* Added more test cases

zhuleyan committed:
display error message (#204)

shumarb committed:
Edited variable name for duplicate check (#203)

* Edited variable name

* Updates based on Travis

* Added back dataencryption tag

zhuleyan committed:
Specify the import command (#205)

* specify the import command

* rephrasing

davidten committed:
Set office location for Google Maps (#207)

* Sets the user location for Google Map use later

davidten committed:
Fixes the window title of linkedin login (#208)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

A0155428B committed:
Improves change theme command (#210)

* Changes implementation of ChangeTheme command to enable dynamic change theme
Adds a default blue theme

* correts style
annonate unused code

* corrects style

* Update default theme in GuiSettings

* Update change theme command test with new implementation

WoodyLau committed:
Fix editdetails (#211)

zhuleyan committed:
improve dev guide and add reused annotation (#214)

shumarb committed:
Author annotation of team members for collation (#212)

* Author annotation of team members for collation

* Added author annotation for team member + edited find command message

* Edited find command message

* Re-instated original message

* Edit find command message

* Edited find command message

* Edited annotation

zhuleyan committed:
Add Portfolio (#216)

* portfolio

zhuleyan committed:
Enables it to tell users the index of person in csv file who has wrong format (#217)

* display more specific error message

* make variable name easier to understand

* change variable name

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

shumarb committed:
Personal Portfolio (#197)

* Updated portfolio

shumarb committed:
Added 2 test cases for display command (#220)

davidten committed:
Fix save config bug (#218)

* fixes config overwrite

shumarb committed:
Edited DeveloperGuide (#222)

shumarb committed:
Personal Portfolio updates (#223)

* Added links to my portions of Developer Guide

* Edited name to my full name on IVLE

* Updated my name and responsibilities

* Updated my name that is reflected on IVLE

* Update meaning of basic command, replaced + with ,

* Replaced old UI with new UI

WoodyLau committed:
Portfolio (#213)

WoodyLau committed:
Add Account command (#225)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
personal portfolio

davidten committed:
remove personal portfolio in master

davidten committed:
Fix user guide section 2 missing (#227)

davidten committed:
Google maps on display if office location set (#224)

* Show Google Maps if the home office location is set

davidten committed:
Personal portfolio (#228)

* Personal Portfolio

A0155428B committed:
Updates autocomplete for v1.5rc (#230)

davidten committed:
UserGuide (#209)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

A0155428B committed:
annotate change theme command tests

A0155428B committed:
Merge pull request #231 from A0155428B/Annotation

annotate change theme command tests

A0155428B committed:
Documentation1.5rc (#229)

* Updates UG to indicate the themes present
Updates acknoledgement for auto-complete

* update UG for blue theme

* Updates portfolio in AboutUs
Updates user story of change theme command in DG
Updates change theme command in UG

* adds new line at EOF

* updates DG to inclue source of code snip

* adds example to change command in UG

* updates DG and portfolio

* update contribution

* adds in contribution to DG header

shumarb committed:
Collated code (#232)

zhuleyan committed:
change annotation

A0155428B committed:
removes whitespace from portfolio

shumarb committed:
Merge pull request #235 from zhuleyan/reuse

Change annotation

Commit:
Collated code for v1.5

A0155428B committed:
removes whitespace from portfolio (#237)

zhuleyan committed:
Merge pull request #238 from Sheikh-Umar/v1.5-code-collation

Code Collation

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Fix linkedin cannot share after logging in (#239)

* personal portfolio

* remove personal portfolio in master

* bug fix

A0155428B committed:
Updates the change theme command in DG

A0155428B committed:
resize sequence diagram

A0155428B committed:
resize high level sequence diagram

shumarb committed:
Personal Portfolio (#241)

* Copy-pasted portions of UserGuide & DeveloperGuide

* Updated name

* Updated description of code collation

* Updated code collation description

* Followed johndoe.adoc style

* Elaborated more on minor enhancements

A0155428B committed:
adds meaningful tags for sample data (#242)

* removes whitespace from portfolio

* adds meaningful tags

* Update test for color tag

* updates GuiTestAssert to accomodate change in sample data tags

WoodyLau committed:
Fix portfolio links (#243)

* Update features for Woody

* Add short description of Leads and Contacts feature

* add build.gradle

* Add convert Use Case. Add collating

* Add expected commands, add feature descriptions

* Fix duplicate import

* Fix mistaken deletion

* Fix MSS

* Add editdetails to User Guide

* Add editdetails to command summary

* Update responsibilities

* Account command will be implemented in 1.5, for now use editdetails

* Change user guide after feedback

* Add collation, tags to docs, create portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Flesh out Developer Guide

* Fix trailing whitespace

* Add documentation changes

* Add account command to things I've done

* Fix links in portfolio

WoodyLau committed:
Fix Title not going through Converts (#244)

* Add Ui image

* Implement Leads and Contacts

* Fix style problems

* Fix style problems

* Revert syntax changes, resolve conflict probably

* Add comments, fix converting contact bugs

* Update documentation and fix bugs

* Fix style

* Fix style

* Added editdetails command, which lets user edit nonessential fields for Leads and Contacts

* Fix some style

* Fix some styles

* Fix some styles

* Fix some styles

* Fix some styles

* Added details fields to UI

* Fix changes

* Fix changes

* Fix multiword companies in Editdetails, set up Adddetails that is an alias for Editdetails

* Fix lack of error message when Account is wrong

* Add account command

* Fix example and style

* Fix example

* Fix style

* Make title pass through converts

zhuleyan committed:
Portfolio (#246)

* portfolio

* new line at EOF

* add portfolio

* edit portfolio

* add johedoe back

* link to my part

* edit link

* delete line

* update

* update

* edit link

* change portfolio

A0155428B committed:
corrects typo

A0155428B committed:
corrects typo in protfolio

A0155428B committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F11-B1/main

* 'master' of https://github.com/CS2103JAN2018-F11-B1/main:
  Portfolio (#246)
  Fix Title not going through Converts (#244)
  Fix portfolio links (#243)
  adds meaningful tags for sample data (#242)
  Personal Portfolio (#241)
  Fix linkedin cannot share after logging in (#239)
  removes whitespace from portfolio (#237)
  Collated code for v1.5
  change annotation

zhuleyan committed:
Merge pull request #245 from A0155428B/documentationfinal

Updates the change theme command in DG

A0155428B committed:
adds coveralls badge

zhuleyan committed:
Merge pull request #247 from A0155428B/coveralls

enables Coveralls

A0155428B committed:
updates protfolio
provides short explanation for unused code

zhuleyan committed:
Merge pull request #248 from A0155428B/master

documentation changes

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Documentation changes (#250)

* personal portfolio

* remove personal portfolio in master

* update tag

* change personal portfolio

shumarb committed:
Add Parent Company Tag feature (#251)

* Updated description of 2nd minor feature

* Added action diagram of parent company tag feature

* Added v2.0 feature: Parent Company tag

* Removed spacing

shumarb committed:
Updates to Personal Portfolio (#249)

* Copy-pasted portions of UserGuide & DeveloperGuide

* Updated name

* Updated description of code collation

* Updated code collation description

* Followed johndoe.adoc style

* Elaborated more on minor enhancements

* Edited description of CRM

zhuleyan committed:
add sort and import command testing

zhuleyan committed:
change tag back

Commit:
Edited tag

zhuleyan committed:
remove unused part

Commit:
Added my portion to portfolio, changed 'action' to activity

Commit:
Changed action to activity

Commit:
Changed naming of diagram

davidten committed:
Linkedin share bug (#252)

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
corrects tag

Commit:
Updated activity diagram

Commit:
Updated description of activity diagram

Commit:
Updated duplicate detection activity diagram

zhuleyan committed:
Merge pull request #253 from zhuleyan/sort-import-testing

Add testing

Commit:
Added use case for parent company tag

Commit:
Added portion of MSS for add parent company tag

Commit:
Added v2.0 label

Commit:
Edited description of README.adoc

Commit:
Updated tagging

Commit:
Updated tags

Commit:
Added colon

Commit:
Added colon

Commit:
Edited position of tags

shumarb committed:
Merge pull request #254 from Sheikh-Umar/v1.5-documentation-updates

Documentation updates

zhuleyan committed:
Add Manual testing and modify import command (#258)

* modify the import command and add manual testing

* new line at EOF

* remove trailing whitespace

* remove unused import

* remove unused thing

shumarb committed:
Documentation updates (#255)

* Updated description

* Updated description

* Removed spacing and unexplained data encryption portion

* Removed spacing

* Changed addressbook.jar to crmbook.jar

* Added link to release

* Added manual testing for my features

* Updated find to Find

* Added testing for alias commands

* Added manual testing portion for display command

* Added manual testing for duplicate detection

* Added manual testing for alias commands

* Added manual testing for find command

* Removed survey

* Removed s

davidten committed:
Refactor ShareToLinkedIn Page

WoodyLau committed:
Added Manual testing instructions (#256)

WoodyLau committed:
Fix the deleting of fields from editdetails (#257)

* Add Ui image

* Implement Leads and Contacts

* Fix style problems

* Fix style problems

* Revert syntax changes, resolve conflict probably

* Add comments, fix converting contact bugs

* Update documentation and fix bugs

* Fix style

* Fix style

* Added editdetails command, which lets user edit nonessential fields for Leads and Contacts

* Fix some style

* Fix some styles

* Fix some styles

* Fix some styles

* Fix some styles

* Added details fields to UI

* Fix changes

* Fix changes

* Fix multiword companies in Editdetails, set up Adddetails that is an alias for Editdetails

* Fix lack of error message when Account is wrong

* Add account command

* Fix example and style

* Fix example

* Fix style

* Make title pass through converts

* Fix removing of editdetails fields

* Fix style

davidten committed:
travis style

davidten committed:
check style

WoodyLau committed:
Add test for editdetails (#259)

* Update features for Woody

* Add short description of Leads and Contacts feature

* add build.gradle

* Add convert Use Case. Add collating

* Add expected commands, add feature descriptions

* Fix duplicate import

* Fix mistaken deletion

* Fix MSS

* Add editdetails to User Guide

* Add editdetails to command summary

* Update responsibilities

* Account command will be implemented in 1.5, for now use editdetails

* Change user guide after feedback

* Add collation, tags to docs, create portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Flesh out Developer Guide

* Fix trailing whitespace

* Add documentation changes

* Add account command to things I've done

* Fix links in portfolio

* Added manual testing

* Note that company cannot be deleted

* Add converteddate to convert test. Add editdetails test

* Add collation. Fix extra file changes

* Fix style

* Fix style

davidten committed:
travis and refactor

davidten committed:
more code refactor

davidten committed:
code quality

davidten committed:
travis

davidten committed:
refactor config

davidten committed:
remove unused imports

davidten committed:
explain decrypter logic in the dev guide

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master' into code-quality

davidten committed:
travis

davidten committed:
OAuth2Client Test

davidten committed:
remove stuff

davidten committed:
remove unused import statement

davidten committed:
remove whitespace

davidten committed:
line length

davidten committed:
travis and simple decrypter test

davidten committed:
travis

davidten committed:
change import order

davidten committed:
Add test coverage for setting location

davidten committed:
add code annotation

davidten committed:
add code credit

davidten committed:
travis and remove johndoe

davidten committed:
clears config before running tests

davidten committed:
travis

davidten committed:
travis

davidten committed:
readd testcases

davidten committed:
travis

davidten committed:
Add manual testing guide

davidten committed:
remove semicolons from manual testing guide for consistency

shumarb committed:
Merge pull request #260 from davidten/code-quality

David code quality improvements

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master' into code-quality

davidten committed:
Remove unnecessary event call

davidten committed:
add dev guide to posting to LinkedIn

davidten committed:
remove unused imports

davidten committed:
add in sequence diagram for LinkedInShare

davidten committed:
update diagram

davidten committed:
add design considerations and put into davidten.adoc

davidten committed:
grammar

davidten committed:
setLocationparser test

davidten committed:
travis

davidten committed:
add setLocation dev guide

davidten committed:
add image descriptor

davidten committed:
Dev guide for loading Google Maps

davidten committed:
add logging statements

A0155428B committed:
adds GuiThemeSettingTest (#262)

A0155428B committed:
adds testing instruction for change theme (#263)

davidten committed:
change sample data for demo suitability

shumarb committed:
Merge pull request #261 from davidten/code-quality

David Code quality part 2

davidten committed:
doc

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master' into dchanges

davidten committed:
user guide

davidten committed:
change contact us adoc

WoodyLau committed:
Fix userguide (#264)

* Update features for Woody

* Add short description of Leads and Contacts feature

* add build.gradle

* Add convert Use Case. Add collating

* Add expected commands, add feature descriptions

* Fix duplicate import

* Fix mistaken deletion

* Fix MSS

* Add editdetails to User Guide

* Add editdetails to command summary

* Update responsibilities

* Account command will be implemented in 1.5, for now use editdetails

* Change user guide after feedback

* Add collation, tags to docs, create portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Editing links in portfolio

* Flesh out Developer Guide

* Fix trailing whitespace

* Add documentation changes

* Add account command to things I've done

* Fix links in portfolio

* Added manual testing

* Note that company cannot be deleted

* Fix style

* Removed past version

* Fix merge

davidten committed:
more doc changes

davidten committed:
Merge branch 'master' into dchanges

shumarb committed:
Merge pull request #265 from davidten/dchanges

Fix documentations

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
Merge remote-tracking branch 'CS2103JAN2018-F11-B1/master'

davidten committed:
better error message, code collation

shumarb committed:
Merge pull request #266 from davidten/master

Better error message

shumarb committed:
Code Collation (#267)

* Functional and Test code collation

* Remove previous unused code

* Latest unused code
